packages/testcontainers/__tests__/containers/NativeChainContainer/NativeChainContainerRestart.test.ts

```diff
@@ -0,0 +1,115 @@
+import { GenesisKeys, NativeChainContainer, StartedNativeChainContainer } from '../../../src'
+// import { RegTestFoundationKeys } from '@defichain/jellyfish-network'
+// import { TestingGroup } from '@defichain/jellyfish-testing'
+import { Network } from 'testcontainers'
+
+describe('nativechain container restart with setDeFiConf', () => {
+  let container: StartedNativeChainContainer
+
+  beforeEach(async () => {
+    const startedNetwork = await new Network().start()
+    const genericContainer = new NativeChainContainer()
+      .withNetworkMode((startedNetwork).getName())
+      .withPreconfiguredRegtestMasternode()
+      .withStartupTimeout(180_000)
+    container = await genericContainer.start()
+    await container.waitForWalletCoinbaseMaturity()
+  })
+
+  afterEach(async () => {
+    await container.stop()
+  })
+
+  it('should create a masternode and mint some blocks with setDeFiConf masternode_operator=', async () => {
```

Comment:
In order to test if restart bahviour is working as expected, you'll have to use a testing group for this. Here since you are testing already with a masternode, setting it to a masternode doesn't really change anything. What you are essentially doing here is making a master node a master node again.

Ideally you would want a test case where you have 2 nodes in a network, with one of them being a masternode, and the other not a masternode. And you'll set the non masternode to a masternode and check if it works after restart. You can refer to https://github.com/JellyfishSDK/jellyfish/blob/main/packages/jellyfish-api-core/__tests__/category/masternode/getMasternodeBlocks.test.ts which shows how its done using TestingGroup.

There's no need to test the restart behaviour for the NativeChainContainer, as it uses the implementation from `testcontainers` already, which is tested by them.

Reply:
cc @infinia-yzl @delphk

Reply:
Will resolve this in a separate issue.